### ISSUE TITLE
a trivial issue: sbcl can't find its default core (partially because of my peculiar installation)

### ISSUE THREAD

**Issue by mifigiel** [joaotavora/sly#310]
dramatis personae:
- `GNU Emacs 28.0.50`
- `SBCL 2.0.3`
- `SLY 1.0.0-beta-3 (#<MREPL mrepl-1-1>)`

my sbcl lisp installations live each in its own  `/home/user/sbcl/<version>`  directory and I've got a symlink in `/home/user/bin/` pointing to my currently preferred sbcl version. 
My `sly-lisp-implementations` is set to `'((sbcl "/home/user/bin/sbcl") (ccl "/home/user/bin/ccl"))` .

- in the shell, I call `sbcl` and get the REPL. 
- in Emacs, sly fails with `"Failed to connect to Slynk: inferior process exited."`

The root cause is sbcl complaining about not being able to find the core relative to `/home/user/bin/sbcl`. I resolved this in `sly-start-lisp` by replacing the `program` in the `comint-exec` call with `(file-truename (executable-find program))`.  

I'm pretty sure that my configuration, as described above, worked in Emacs 27.0.50 and it absolutely did work in 26.3, so I guess it's a change in how `comint` deals with the program name.

And last but not least: thank you for sly: it's awesome!

**Comment by joaotavora** [comment]
> dramatis personae:

lol, nice touch.

I still have to move on to Emacs 28 (master). I think at least some things are backward incompatible to current sly.  Will try it this week.

> The root cause is sbcl complaining about not being able to find the core relative to /home/user/bin/sbcl. I resolved this in sly-start-lisp by replacing the program in the comint-exec call with (file-truename (executable-find program))

Read this very briefly, but in the meantime have you tried fiddling with `default-directory`?  Or, simpler, passing the full path name to the core file?  Or using `run-sbcl.sh`?

**Comment by mifigiel** [comment]
> Read this very briefly, but in the meantime have you tried fiddling with `default-directory`? Or, simpler, passing the full path name to the core file? Or using `run-sbcl.sh`?

Yes,  a shell script ( `exec sbcl` ) solves it nicely. It's independent of sly/emacs and I can keep following my twisted ways - great!
